kernel/core/utils/stringtools.hpp

```diff
@@ -130,6 +131,241 @@ enum TimeDateFormat
+    int endParseNumber(int idx)
+    {
+        // we only want to check for the last 2/1
+        m_tape = m_tape.back();
+
+        pushInbetween(idx); //since normally the function is called when seperator appears
+        m_curr_format |= checkNumFormat(m_tape);
+
+        m_last_idx = idx;
+        m_tape = "";
+        return m_curr_format;
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function is called to make a vote with the given number Type
+    ///
+    /// \param numType int
+    /// \return void
+    ///
+    /////////////////////////////////////////////////
+    void vote(int numType)
+    {
+        if((numType & NUM_DECIMAL_EU && numType & NUM_K_US) ||
+                (numType & NUM_DECIMAL_US && numType & NUM_K_EU))
+        {
+            m_num_format_votes[FMT_INVALID]++;  // INVALID Combination
+        }
+        else
+        {
+            if(numType & NUM_DECIMAL_EU)
+                m_num_format_votes[FMT_DEC_EU]++;
+            else if(numType & NUM_DECIMAL_US)
+                m_num_format_votes[FMT_DEC_US]++;
+
+            if(numType & NUM_K_EU)
+                m_num_format_votes[FMT_K_EU]++;
+            else if(numType & NUM_K_US)
+                m_num_format_votes[FMT_K_US]++;
+            else if(numType & NUM_K_SPACE)
+                m_num_format_votes[FMT_K_SPACE]++;
+
+            if(numType & NUM_INVALID)
+                m_num_format_votes[FMT_INVALID]++;
+        }
+    }
+
+    /////////////////////////////////////////////////
+    /// \brief This function returns the current Format given by votes
+    ///
+    /// \return int
+    ///
+    /////////////////////////////////////////////////
+    int getFormat()
+    {
+        // As soon as there are more than 2 set, we will have at least one bad sample that would be parsed wrong
+        int inval_cnt = 0;
+        for(int i = 0; i < FMT_COUNT; i++)
+        {
+            if(m_num_format_votes[i] > 0)
+                inval_cnt++;
+        }
+        if(inval_cnt > 2 || m_num_format_votes[FMT_INVALID] > 0)
+            return NUM_INVALID;
+
+        int num_format = 0;
+        if((m_num_format_votes[FMT_K_EU] > m_num_format_votes[FMT_K_US]) && (m_num_format_votes[FMT_K_EU] > m_num_format_votes[FMT_K_SPACE]))
+            num_format |= NUM_K_EU;
+        else if((m_num_format_votes[FMT_K_US] > m_num_format_votes[FMT_K_EU]) && (m_num_format_votes[FMT_K_US] > m_num_format_votes[FMT_K_SPACE]))
+            num_format |= NUM_K_US;
+        else if(m_num_format_votes[FMT_K_SPACE] > 0)
+            num_format |= NUM_K_SPACE;
+
+        int add = 0;
+        if(num_format == 0)
+            add = 1;
+
+        if(m_num_format_votes[FMT_DEC_EU] > m_num_format_votes[FMT_DEC_US])
+            num_format |= NUM_DECIMAL_EU | (NUM_K_EU * add);
+        else if(m_num_format_votes[FMT_DEC_US] > 0)
+            num_format |= NUM_DECIMAL_US | (NUM_K_US * add);
+
+        return num_format;

```

Comment:
You always assume, there's a strong order within your votes, which might not be true: `1 000 000,00; 1,000,000.00`. Each format now appears the same time, so it might happen that `num_format` is still 0 at this point. What does 0 mean? Isn't that just the same thing as `NUM_INVALID`?

Reply:
Good Point!
I did assume that we want to return NUM_INVALID when two Formats accure in the > 2 check before.

`num_format ` = 0 would mean no unique Number Format found, but also no INVALID combination did appear. 
This means, the convertion Function will do nothing. This would happen if all nums are without any seperators at all, or all are 10.000  or 1,000 where it could not be specified if thousans or decimal seperator.

Reply:
OK, but isn't that in the end the same thing (for conversion) as `NUM_INVALID`. Semantically speaking, no format is also an invalid format.

Regarding the last point: if you cannot decide, whether EU or US, the default should be the US format for numbers.

Reply:
Returning `NUM_INVALID` should set `CONVTYPE_NONE`, then no conversion is done. 
When 0 is returned `strChangeNumberFormat(...)`, would currently do nothing and `StrToCmplx(...)` is called with the string as it is. 

But I guess we should not return 0, but US format right? 
We should only return 0 in two cases:
- No separators at all, so no conversation needed.
- We only find separators ',' or '.' with no more than 3 digits left and exactly 3 digits right. (eg. 100.000 or 10,000)

Reply:
Following my previous comment, I'd appreciate if it defaults to US format but allows for EU format together, if none of them are using their thousands separator. If that is the same thing as return 0 in your implementation (with the addition that `replaceall(STR, ",", ".")` is called), then that's the way to go.